app/src/main/java/com/example/lwilcox/lab2/StreamFragment.java

```diff
@@ -0,0 +1,139 @@
+                AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getActivity());
+                alertDialogBuilder.setMessage(context.getResources().getString(R.string.delete));
+                alertDialogBuilder.setPositiveButton(context.getResources().getString(R.string.yes), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface arg0, int arg1) {
+                        FeedReaderDBH.removeItem(imageLinks.get(photoIndex));
+                        imageLinks = FeedReaderDBH.readList();
+                        setImage();
+                    }
+                });
+                alertDialogBuilder.setNegativeButton(context.getResources().getString(R.string.nope),
+                        new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                    }
+                });
+                AlertDialog alertDialog = alertDialogBuilder.create();
+                alertDialog.show();
+            }
+        });
+        return myFragmentView;
+    }
+    public void setImage(){
+        if (imageLinks.size() > 0) { //prevents the app from crashing if the stream page is hit without any saved images
+            if (photoIndex < imageLinks.size()) { //if photoIndex is still a valid number okay
+                String thumbnailLink = imageLinks.get(photoIndex);
+                new ImageLoadTask(thumbnailLink, imageView).execute();
+            }
+            else{ //otherwise bring the index down one to account for the deleted entry
+                String thumbnailLink = imageLinks.get(photoIndex - 1);
+                new ImageLoadTask(thumbnailLink, imageView).execute();
+            }
+        }
+        else{
+            AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getActivity());
+            alertDialogBuilder.setMessage(getContext().getResources().getString(R.string.null_to_search));
+            alertDialogBuilder.setPositiveButton(getContext().getResources().getString(R.string.okay),
+                    new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface arg0, int arg1) {
+                            //this is blank because if I put switchFragments here people can finagle
+                            //their way out of clicking the okay button and if they don't leave the
+                            //page it will crash
+                        }
+                    });
+            AlertDialog alertDialog = alertDialogBuilder.create();
+            alertDialog.show();
+            switchFragments(); //so no matter what they get kicked off the stream page
+        }
+    }
+    public void switchFragments(){
+        FragmentTransaction ft = getActivity().getSupportFragmentManager().beginTransaction();
```

Comment:
Maybe this switching fragments code could also be in a method so you don't have to duplicate it 3 times.